### ISSUE TITLE
More elaborate classification output

### ISSUE THREAD

**Issue by hslh** [simondewit/pickleteam#7]
For example, show performance per class (emoji), and some kind of confusion matrix. This way, you can see which classes get confused the most (and whether that makes sense), and if the classifier does well over all classes, or whether performance varies a lot per class.

**Comment by hslh** [comment]
Also add performance (precision/recall/F1) per class

**Comment by hslh** [comment]
For getting the labels from Keras, running ```np.argmax(predictions, axis=1)``` should do the trick, where ```predictions``` is the output of ```model.predict```

**Comment by DaphneGroot** [comment]
Extracting the labels is done, but when using some code I found online to calculate the metrics per class, I noticed that the precision, recall and fscores were different from what we previously found.
The scores per class are obtained using `classification_report(test_categories, predicted_categories) ` (from sklearn, implemented in the keras file)
Because of that, I tried to calculate the precision, recall and F-score using sklearn as well, instead of using the functions used before.
It turns out that the scores do differ, but which one will be the right one?

The file with the "original" calculation of the precision, recall and F-score is kerasScoresPerClass2.py, the one with the sklearn code is kerasScoresPerClass1.py

**Comment by hslh** [comment]
Interesting! I'll have a look. In either case, I'd say the correct one is the one that gives the same numbers as the evaluation script provided by the organizers at [https://competitions.codalab.org/competitions/17344#learn_the_details-evaluation](https://competitions.codalab.org/competitions/17344#learn_the_details-evaluation) . Can you implement that one, and see what that outputs?

**Comment by hslh** [comment]
I think the one in  kerasScoresPerClass2 is incorrect, but I don't really get what it's doing. Either way, the sklearn metrics are probably more reliable, but we can confirm that by comparing to the official evaluation. In any case, make sure to use ```average="macro"``` in the sklearn metrics, since we are concerned about macro-averaged F1-score here.

**Comment by hslh** [comment]
Also, don't forget to change the evaluation so that it runs on only the development data, rather than also part of the training data.

**Comment by DaphneGroot** [comment]
The one from kerasScoresPerClass1 is indeed the correct one. The f-scores from that file are the same as the one from the evaluation script provided by the organizers.

The only problem we still have, and we have no idea why, is that we get no results (all zero's when using the English data) when using the development data as test data. I've created the new file (kerasScoresPerClass.py) with the correct metrics per class and the new way of splitting training and testing data. Do you have an idea why we get no results when using this?

**Comment by hslh** [comment]
Just by looking at the code, I don't. I could run it, but I don't have the development data (```tweetsAsTuplesFileTrialEnglish.pickle```). Could you put that on Github? Then I can have a look.

**Comment by DaphneGroot** [comment]
Oops, sorry! I've uploaded them!
